linux/libcmt/include/libcmt/buf.h

```diff
@@ -0,0 +1,75 @@
+/** @file
+ * @defgroup libcmt_buf buf
+ * slice of contiguous memory
+ *
+ * @ingroup libcmt
+ * @{ */
+#ifndef CMT_BUF_H
+#define CMT_BUF_H
+#include<stdint.h>
+#include<stddef.h>
+
+/** A slice of contiguous memory from @b begin to @b end.
+ * Size can be taken with: `q - p`.
+ *
+ * `begin == end` indicate an empty buffer */
+typedef struct {
+	uint8_t *begin; /**< begin of memory region */
+	uint8_t *end;   /**< end of memory region */
+} cmt_buf_t;
+
+/** Declare a cmt_buf_t with stack backed memory.
+ * @param [in] N - size in bytes
+ * @note don't port */
+#define CMT_BUF_DECL(S, L) cmt_buf_t S[1] = {{ \
```

Comment:
When I run `clang-tidy` I get this error:

```
/home/bart/projects/cartesi/machine-emulator-sdk/fs/external/tools/linux/libcmt/tests/abi.c:155:2: warning: The left operand of '+' is a garbage value [clang-analyzer-core.UndefinedBinaryOperatorResult]
        CMT_BUF_DECL(b, 64);
        ^
./include/libcmt/buf.h:26:22: note: expanded from macro 'CMT_BUF_DECL'
        .end   = (S)->begin + L,               \
                            ^
```

I think these macros may not work as expected everywhere. I would vote to remove them, and make tests, examples and everything use the non macro version.